cms/lib/xblock/upstream_sync.py

```diff
@@ -297,7 +297,11 @@ def _update_non_customizable_fields(*, upstream: XBlock, downstream: XBlock) ->
     """
     syncable_fields = _get_synchronizable_fields(upstream, downstream)
     customizable_fields = set(downstream.get_customizable_fields().keys())
+    isVideoBlock = downstream.usage_key.block_type == "video"
     for field_name in syncable_fields - customizable_fields:
+        if isVideoBlock and field_name == 'edx_video_id':
+            # Avoid overwriting edx_video_id between blocks
+            continue
```

Comment:
@ChrisChV @pomegranited With all due appreciation to the VideoBlock nonsense you've had to grapple with to support transcripts, it truly pains me to see such block-type-specific code in this highly generalized function 😓 Upstream syncing is complex but nature, but adding special cases like this move it from being "complex" to being "complicated".

Is there any other way we can ensure that edx_video_id isn't synced? Would it not work to put it in the dict returned by `get_customizable_fields`, as @pomegranited did for `graceperiod`, `grading_method`, etc.?

Reply:
+1, good point

Though the two root problems here are (1) that there is no standard way to associate a block with external data that's not in an XBlock field (e.g. edx_video_id points to VAL or ORA2 has tons of Django models holding its data) and (2) that there is no proper XBlock API for dealing with static files (e.g. transcripts)